File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Usando o operador de espalhamento/Código em Python/spreadOperator.py

# Usaremos os dois conjuntos a seguir em nossos exemplos:
setA = set()
setA.add(1)
setA.add(2)
setA.add(3)
setB = set()
setB.add(2)
setB.add(3)
setB.add(4)

# Union
print({ *setA, *setB }) #  {1, 2, 3, 4}

# Intersection
print({ x for x in {*setA} if x in {*setB} }) # {2, 3}

# Difference
print({x for x in {*setA} if x not in {*setB}}) # {1}


# Também é possível realizar essa operacções com os operadores: |, & e -
# Union
# Usando o operador |
print(setA | setB) #  {1, 2, 3, 4}

# Intersection
# Usando o operador &
print(setA & setB) # {2, 3}

# Difference
# Usando o operador -
print(setA - setB) # {1}

File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações com a classe Set da ES2015/Usando o operador de espalhamento/spreadOperator.js
// Usaremos os dois conjuntos a seguir em nossos exemplos:
const setA = new Set();
setA.add(1);
setA.add(2);
setA.add(3);
const setB = new Set();
setB.add(2);
setB.add(3);
setB.add(4)

// Union
console.log(new Set([...setA], [...setB]));

// Intersection
console.log(new Set([...setA].filter(x => setB.has(x))));

// Diferença
console.log(new Set([...setA].filter(x => !setB.has(x))));